wp-security-audit-log.php

```diff
@@ -190,6 +174,23 @@ public function Init()
     {
         // Start listening to events
         WpSecurityAuditLog::GetInstance()->sensors->HookEvents();
+        
+        if ($this->settings->IsArchivingEnabled()) {
+            // Check the current page
+            if (isset($_REQUEST['page']) && $_REQUEST['page'] == 'wsal-auditlog') {
+                // Keeps the transient
+            } else {
+                if (defined('DOING_AJAX') && DOING_AJAX) {
+                    // It's an AJAX call
+                } else {
+                    $selected_db = get_transient('wsal_wp_selected_db');
+                    if ($selected_db) {
+                        // Delete the transient
+                        delete_transient('wsal_wp_selected_db');
+                    }
+                }
+            }
+        }

```

Comment:
@carmelopreeostudios this should be changed to `if (!(isset($_REQUEST['page']) && $_REQUEST['page'] == 'wsal-auditlog'  && defined('DOING_AJAX') && DOING_AJAX))` and then do the following code:

`
$selected_db = get_transient('wsal_wp_selected_db');
    if ($selected_db) {
        // Delete the transient
        delete_transient('wsal_wp_selected_db');
    }
`